src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -738,12 +895,164 @@ export class SavedObjectsRepository {
       id,
       type,
       updated_at: time,
-      version: encodeHitVersion(response),
+      version: encodeHitVersion(updateResponse),
+      ...(this._registry.isMultiNamespace(type) && {
+        namespaces: updateResponse.get._source.namespaces,
+      }),
       references,
       attributes,
     };
   }
 
+  /**
+   * Adds one or more namespaces to a given multi-namespace saved object. This method and
+   * [`deleteFromNamespaces`]{@link SavedObjectsRepository.deleteFromNamespaces} are the only ways to change which Spaces a multi-namespace
+   * saved object is shared to.
+   */
+  async addToNamespaces(
+    type: string,
+    id: string,
+    namespaces: string[],
+    options: SavedObjectsAddToNamespacesOptions = {}
+  ): Promise<{}> {
+    if (!this._allowedTypes.includes(type)) {
+      throw SavedObjectsErrorHelpers.createGenericNotFoundError(type, id);
+    }
+
+    if (!this._registry.isMultiNamespace(type)) {
+      throw SavedObjectsErrorHelpers.createBadRequestError(`${type} doesn't support namespaces`);
```

Comment:
ultra-nit: `doesn't support multi-namespaces` may be a little more explicit?

Reply:
You're right, I'll update the message.

Reply:
Not sure how I missed this on the last pass, but I updated this in 223ecef.